config.cc

```diff
@@ -2221,7 +2256,19 @@ QString getPidFileName() THROW_SPEC( exError )
 
 QString getHistoryFileName() THROW_SPEC( exError )
 {
+  #ifdef Q_OS_LINUX
+    QDir home = getHomeDir();
```

Comment:
1) `const QDir`;
2) Declare this variable above the `#ifdef` and use it everywhere instead of calling `getHomeDir()` repeatedly.